allennlp/models/encoder_decoders/nlvr_semantic_parser.py

```diff
@@ -146,9 +146,46 @@ def forward(self,  # type: ignore
 
         outputs = self._decoder_trainer.decode(initial_state, self._decoder_step,  # type: ignore
                                                self._max_decoding_steps)
-
+        best_action_sequences = outputs['best_action_sequence']
+        get_action_string = lambda rule: "%s -> %s" % (rule["left"][0], rule["right"][0])
+        labels_data = label.data.cpu()
```

Comment:
You should call `.numpy()` here, maybe also `.tolist()`, it should make things faster.  Or maybe because you're calling `.data.cpu()` it's enough...  Hmm, I'm not really sure.